packages/shared/isValidElementType.js

```diff
@@ -54,6 +59,12 @@ export default function isValidElementType(type: mixed) {
       type.$$typeof === REACT_CONTEXT_TYPE ||
       type.$$typeof === REACT_FORWARD_REF_TYPE ||
       type.$$typeof === REACT_FUNDAMENTAL_TYPE ||
+      // This needs to include all possible module reference object
+      // types supported by any Flight configuration anywhere since
+      // we don't know which Flight build this will end up being used
+      // with.
+      type.$$typeof === REACT_MODULE_REFERENCE ||
+      type._moduleId !== undefined ||
```

Comment:
I don't really get this. Why can't we use the brand check? Why can't different systems agree on the Symbol to use if they have to agree on the field name anyway?

Reply:
They don’t have to agree on either. The field check is just because I can’t require JSResourceReference in other environments but it should really be an instanceof check. We just have to add all possible checks here. If this was a true private field we couldn’t check the field neither. That might even mean that we have to relax it to accept all objects.

Reply:
Ugh. This didn't even work because we obscure underscore prefixed properties.